### ISSUE TITLE
panic in `indexed_search.go`

### ISSUE THREAD

**Issue by camdencheek** [sourcegraph/sourcegraph#25600]
I hit a panic locally in `indexed_search.go` on `main`. Have not investigated yet. Looking into it now.

<details>
  <summary>Stacktrace</summary>

```        
        panic: runtime error: invalid memory address or nil pointer dereference
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2736c22]

        goroutine 10192 [running]:
        github.com/sourcegraph/sourcegraph/internal/search/zoekt.(*IndexedRepoRevs).getRepoInputRev(0xc002447b30, 0xc00468a2a0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/zoekt/indexed_search.go:160 +0x142
        github.com/sourcegraph/sourcegraph/internal/search/zoekt.sendMatches(0x0, 0xc002d0ef80, {0x34247e1, 0x6}, {0x3b6cfa0, 0xc001bf73e0})
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/zoekt/indexed_search.go:585 +0x1da
        github.com/sourcegraph/sourcegraph/internal/search/zoekt.zoektSearch.func2(0x108b697)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/zoekt/indexed_search.go:552 +0x79
        github.com/sourcegraph/sourcegraph/internal/search/backend.ZoektStreamFunc.Send(0x0, 0x0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/zoekt.go:32 +0x1f
        github.com/sourcegraph/sourcegraph/internal/search/backend.(*meteredSearcher).StreamSearch.func5(0xc003da80c0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/metered_searcher.go:178 +0x2e4
        github.com/sourcegraph/sourcegraph/internal/search/backend.ZoektStreamFunc.Send(0x3be9bf8, 0xc00181ce58)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/zoekt.go:32 +0x1f
        github.com/sourcegraph/sourcegraph/internal/search/backend.(*HorizontalSearcher).StreamSearch.func1.1(0xc003da80c0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/horizontal.go:120 +0x252
        github.com/google/zoekt/stream.SenderFunc.Send(0x76, 0xc000654cc0)
                /Users/ccheek/go/pkg/mod/github.com/sourcegraph/zoekt@v0.0.0-20210930114328-078fa3e9d530/stream/client.go:43 +0x1f
        github.com/sourcegraph/sourcegraph/internal/search/backend.(*meteredSearcher).StreamSearch.func5(0xc003da80c0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/metered_searcher.go:178 +0x2e4
        github.com/sourcegraph/sourcegraph/internal/search/backend.ZoektStreamFunc.Send(0xc002b40800, 0x2e72ea0)
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/zoekt.go:32 +0x1f
        github.com/google/zoekt/stream.(*Client).StreamSearch(0xc000675218, {0x3bcbb88, 0xc0025b35f0}, {0x3b6aa00, 0xc0019d7860}, 0xc0019d7810, {0x3b73080, 0xc00284fe00})
                /Users/ccheek/go/pkg/mod/github.com/sourcegraph/zoekt@v0.0.0-20210930114328-078fa3e9d530/stream/client.go:89 +0x733
        github.com/sourcegraph/sourcegraph/internal/search/backend.(*meteredSearcher).StreamSearch(0xc002b1b280, {0x3bcbb88, 0xc0026d5170}, {0x3b6aa00, 0xc0019d7860}, 0xc0019d7810, {0x3b71f80, 0xc0023622a0})
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/metered_searcher.go:155 +0x1465
        github.com/sourcegraph/sourcegraph/internal/search/backend.(*HorizontalSearcher).StreamSearch.func1({0xc00005a257, 0xe}, {0x3be9b78, 0xc002191c20})
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/horizontal.go:94 +0x251
        created by github.com/sourcegraph/sourcegraph/internal/search/backend.(*HorizontalSearcher).StreamSearch
                /Users/ccheek/src/sourcegraph/sourcegraph/internal/search/backend/horizontal.go:93 +0x2f8

```
</details>

**Comment by camdencheek** [comment]
cc @tsenart because of #25594

**Comment by camdencheek** [comment]
Reproducible with the query `repo:monorepo test`

**Comment by camdencheek** [comment]
Found the issue. I recently cleared my DB because of some bad state (experiment gone bad), so there were repos (specifically `monorepo-test-6`) that were not listed in the DB, but zoekt still had indexed, so it was sending results back for repos the frontend didn't know about. When trying to look up what repo a match belongs to, we assume the repo is in the set of repos we asked for. This is usually a valid assumption.

This is not an actual issue right now, so I'm closing this. However, if someone looks at this, if (during normal operation) we can't guarantee that all results returned by zoekt are for repos in the db, this might be an issue.
